### ISSUE TITLE
Pangenome, required parameters are missing

### ISSUE THREAD

**Issue by jellyvanderwoude** [bactopia/bactopia#293]
--wf pangenome function in Bactopia 2.0.2 says required parameters are missing when they aren't

> (bactopia) bio-24046:bactopia-folder2 jvanderwoude3$ bactopia --wf pangenome --bactopia /Volumes/JellyExternal/bactopia-folder2
> 2022-03-09 14:13:52:root:INFO - Checking if environment pre-builds are needed, use --verbose to see full details.
> N E X T F L O W  ~  version 21.10.6
> Launching `/Users/jvanderwoude3/miniconda3/envs/bactopia/share/bactopia-2.0.x/main.nf` [elated_bhaskara] - revision: bd75c553f9
> 
> 
> --------------------------------------------------------------------
>    _                _              _         _              _              
>   | |__   __ _  ___| |_ ___  _ __ (_) __ _  | |_ ___   ___ | |___          
>   | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` | | __/ _ \ / _ \| / __|    
>   | |_) | (_| | (__| || (_) | |_) | | (_| | | || (_) | (_) | \__ \        
>   |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|  \__\___/ \___/|_|___/ 
>                             |_|                                            
>   bactopia tools pangenome v2.0.2
>   Pangenome analysis with optional core-genome phylogeny 
> --------------------------------------------------------------------
> Core Nextflow options
>   runName          : elated_bhaskara
>   launchDir        : /Volumes/JellyExternal/bactopia-folder2
>   workDir          : /Volumes/JellyExternal/bactopia-folder2/work
>   projectDir       : /Users/jvanderwoude3/miniconda3/envs/bactopia/share/bactopia-2.0.x
>   userName         : jvanderwoude3
>   profile          : standard
>   configFiles      : /Users/jvanderwoude3/miniconda3/envs/bactopia/share/bactopia-2.0.x/nextflow.config
> 
> Required Parameters
>   bactopia         : /Volumes/JellyExternal/bactopia-folder2
> 
> Optional Parameters
>   outdir           : /Volumes/JellyExternal/bactopia-folder2
>   run_name         : pangenome
> 
> Nextflow Profile Parameters
>   condadir         : /Users/jvanderwoude3/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs
>   registry         : quay
>   singularity_cache: /Users/jvanderwoude3/.bactopia/singularity
> 
> Helpful Parameters
>   wf               : pangenome
> 
> !! Only displaying parameters that differ from the pipeline defaults !!
> --------------------------------------------------------------------
> If you use bactopia for your analysis please cite:
> 
> * Bactopia
>   https://doi.org/10.1128/mSystems.00190-20
> 
> * The nf-core framework
>   https://doi.org/10.1038/s41587-020-0439-x
> 
> * Software dependencies
>   https://bactopia.github.io/acknowledgements/
> --------------------------------------------------------------------
> WARN: 
> WARN: Conda Disclaimer
> WARN: 
> WARN: If you have access to Docker or Singularity, please consider
> WARN: running Bactopia using containers. The containers are less
> WARN: susceptible to Conda environment related issues (e.g. version
> WARN: conflicts) and errors caused by creation of conda environments
> WARN: in parallel (use '--max_cpus 1' to over come this error).
> WARN: 
> WARN: To use containers, you can use the profile parameter
> WARN:     Docker: -profile docker
> WARN:     Singularity: -profile singularity
> WARN: 
> --------------------------------------------------------------------
> Required parameters are missing, please check and try again.
> 
> 
> Required Parameters
>   --bactopia                          [string]  The path to bactopia results to use as inputs
> 
> --------------------------------------------------------------------
> ERROR: Validation of pipeline parameters failed!
> Please correct to continue
>

**Comment by rpetit3** [comment]
Hi @jellyvanderwoude!

 I goofed on a parameter requirement (its the traits one). I've fixed it here: https://github.com/bactopia/bactopia/commit/e1d22cb8af07ebc52d64e5717bb882ba21be35f9

The fix is available on my dev-build, and will be in the next version.

```
conda create -n bactopia-dev -c rpetit3 -c conda-forge -c bioconda rpetit3::bactopia
```

This should let you keep moving along, let me know if not!

Cheers,
Robert

**Comment by jellyvanderwoude** [comment]
Thanks Robert! I moved ahead using Bactopia-dev and this was the output

(bactopia-dev) bio-24046:bactopia-folder2 jvanderwoude3$ bactopia --wf pangenome --bactopia /Volumes/JellyExternal/bactopia-folder2 --include include_jorth.txt
N E X T F L O W  ~  version 21.10.6
Launching `/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/main.nf` [reverent_bernard] - revision: 81a0ab6755


--------------------------------------------------------------------
   _                _              _         _              _              
  | |__   __ _  ___| |_ ___  _ __ (_) __ _  | |_ ___   ___ | |___          
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` | | __/ _ \ / _ \| / __|    
  | |_) | (_| | (__| || (_) | |_) | | (_| | | || (_) | (_) | \__ \        
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|  \__\___/ \___/|_|___/ 
                            |_|                                            
  bactopia tools pangenome v2.0.0
  Pangenome analysis with optional core-genome phylogeny 
--------------------------------------------------------------------
Core Nextflow options
  runName          : reverent_bernard
  launchDir        : /Volumes/JellyExternal/bactopia-folder2
  workDir          : /Volumes/JellyExternal/bactopia-folder2/work/pangenome
  projectDir       : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x
  userName         : jvanderwoude3
  profile          : standard
  configFiles      : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/nextflow.config

Required Parameters
  bactopia         : /Volumes/JellyExternal/bactopia-folder2

Filtering Parameters
  include          : include_jorth.txt

Optional Parameters
  outdir           : /Volumes/JellyExternal/bactopia-folder2
  run_name         : pangenome

Nextflow Profile Parameters
  condadir         : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs
  singularity_cache: /Users/jvanderwoude3/.bactopia/singularity

Helpful Parameters
  wf               : pangenome

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
WARN: 
WARN: Conda Disclaimer
WARN: 
WARN: If you have access to Docker or Singularity, please consider
WARN: running Bactopia using containers. The containers are less
WARN: susceptible to Conda environment related issues (e.g. version
WARN: conflicts) and errors caused by creation of conda environments
WARN: in parallel (use '--max_cpus 1' to over come this error).
WARN: 
WARN: To use containers, you can use the profile parameter
WARN:     Docker: -profile docker
WARN:     Singularity: -profile singularity
WARN: 
--------------------------------------------------------------------

Each task will use 4 CPUs out of the available 28 CPUs. At most 
7 task(s) will be run at a time, this can affect the efficiency 
of Bactopia. You can use the '-qs' parameter to alter the number of 
tasks to run at a time (e.g. '-qs 2', means only 2 tasks or a maximum 
of 8 CPUs will be used at once)


Including 91 samples for analysis
CF1_LeftUpper_A7 does not appear to be a Bactopia sample, skipping...
CF1_Lingula_C4 does not appear to be a Bactopia sample, skipping...
CF1_Lingula_C5 does not appear to be a Bactopia sample, skipping...
CF1_RightMiddle_F1 does not appear to be a Bactopia sample, skipping...
CF1_RightMiddle_F6 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_G1 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_H2 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_H3 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_H4 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_H5 does not appear to be a Bactopia sample, skipping...
CF1_RightUpper_H8 does not appear to be a Bactopia sample, skipping...
Found 80 samples to process

If this looks wrong, now's your chance to back out (CTRL+C 3 times).
Sleeping for 5 seconds...
--------------------------------------------------------------------
executor >  local (1)
    Bactopia Tools: `pangenome Execution Summary_TOOL:PIRATE_MODULE (pirate) [100%] 1 of 1 ✔
    ---------------------------OOLS:PANGENOME:SNPDISTS                       -
    Bactopia Version : 2.0.0IATOOLS:PANGENOME:CLONALFRAMEML:IQTREE           -
    Nextflow Version : 21.10.6TOOLS:PANGENOME:CLONALFRAMEML:CLONALFRAME      -
    Command Line     : nextflow run /Users/jvanderwoude3/miniconda3/envs/bactopia-dev//share/bactopia-2.0.x/main.nf -w /Volumes/JellyExternal/bactopia-folder2/work/pangenome --wf pangenome --bactopia /Volumes/JellyExternal/bactopia-folder2 --include include_jorth.txtAREVERSIONS              -
    Resumed          : falseoconda::iqtree=2.1.4_beta [cache /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8]
    Completed At     : 2022-03-09T16:16:28.475855-05:00che /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-e13ce3aa35d34f1f090a292512579027]
    Duration         : 1h 35m 23s
    Success          : false
    Exit Code        : null
    Error Report     : Error executing process > 'BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:IQTREE (start-tree)'

Caused by:
  Failed to create Conda environment
  command: mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8 bioconda::iqtree=2.1.4_beta
  status : 1
  message:
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    
    # >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
    
        Traceback (most recent call last):
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/conda/exceptions.py", line 1080, in __call__
            return func(*args, **kwargs)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 951, in exception_converter
            raise e
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 945, in exception_converter
            exit_code = _wrapped_main(*args, **kwargs)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 901, in _wrapped_main
            result = do_call(args, p)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 767, in do_call
            exit_code = create(args, parser)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 643, in create
            return install(args, parser, "create")
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 507, in install
            index = load_channels(pool, channels, repos)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/utils.py", line 171, in load_channels
            repo = subdir.create_repo(pool)
        RuntimeError: Could not read JSON repodata file (/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache/31ce02e0.json) parse error line 163727
    
    `$ /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/bin/mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8 bioconda::iqtree=2.1.4_beta`
    
      environment variables:
                     CIO_TEST=<not set>
       CONDA_BACKUP_JAVA_HOME=:-:-
    CONDA_BACKUP_JAVA_LD_LIBRARY_PATH=:-:-
            CONDA_DEFAULT_ENV=bactopia-dev
                    CONDA_EXE=/Users/jvanderwoude3/miniconda3/bin/conda
                 CONDA_PREFIX=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev
               CONDA_PREFIX_1=/Users/jvanderwoude3/miniconda3
        CONDA_PROMPT_MODIFIER=(bactopia-dev)
             CONDA_PYTHON_EXE=/Users/jvanderwoude3/miniconda3/bin/python
                   CONDA_ROOT=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev
                  CONDA_SHLVL=2
               CURL_CA_BUNDLE=<not set>
         JAVA_LD_LIBRARY_PATH=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/server
                         PATH=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/bin:/Users/jvanderwo
                              ude3/miniconda3/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:
                              /usr/local/ncbi/blast/bin:/opt/salt/bin:/usr/local/sbin:/opt/X11/bin:/
                              Users/jvanderwoude3/edirect
           REQUESTS_CA_BUNDLE=<not set>
                SSL_CERT_FILE=<not set>
    
         active environment : base
        active env location : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev
                shell level : 2
           user config file : /Users/jvanderwoude3/.condarc
     populated config files : /Users/jvanderwoude3/.condarc
              conda version : 4.11.0
        conda-build version : not installed
             python version : 3.7.12.final.0
           virtual packages : __osx=10.16=0
                              __unix=0=0
                              __archspec=1=x86_64
           base environment : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev  (writable)
          conda av data dir : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/etc/conda
      conda av metadata url : None
               channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                              https://conda.anaconda.org/conda-forge/noarch
                              https://conda.anaconda.org/bioconda/osx-64
                              https://conda.anaconda.org/bioconda/noarch
                              https://repo.anaconda.com/pkgs/main/osx-64
                              https://repo.anaconda.com/pkgs/main/noarch
                              https://repo.anaconda.com/pkgs/r/osx-64
                              https://repo.anaconda.com/pkgs/r/noarch
                              https://conda.anaconda.org/r/osx-64
                              https://conda.anaconda.org/r/noarch
              package cache : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs
                              /Users/jvanderwoude3/.conda/pkgs
           envs directories : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/envs
                              /Users/jvanderwoude3/.conda/envs
                   platform : osx-64
                 user-agent : conda/4.11.0 requests/2.27.1 CPython/3.7.12 Darwin/20.6.0 OSX/10.16
                    UID:GID : 1041747584:536040563
                 netrc file : None
               offline mode : False
    
    
    An unexpected error has occurred. Conda has prepared the above report.


    Launch Dir       : /Volumes/JellyExternal/bactopia-folder2
    
executor >  local (1)
[ad/44ede2] process > BACTOPIATOOLS:PANGENOME:PG_TOOL:PIRATE_MODULE (pirate) [100%] 1 of 1 ✔
[-        ] process > BACTOPIATOOLS:PANGENOME:SNPDISTS                       -
[-        ] process > BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:IQTREE           -
[-        ] process > BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:CLONALFRAME      -
[-        ] process > BACTOPIATOOLS:PANGENOME:FINAL_TREE:IQTREE_MODULE       -
[-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS              -
Creating env using mamba: bioconda::iqtree=2.1.4_beta [cache /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8]
Creating env using mamba: bioconda::snp-dists=0.8.2 [cache /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-e13ce3aa35d34f1f090a292512579027]
Error executing process > 'BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:IQTREE (start-tree)'

Caused by:
  Failed to create Conda environment
  command: mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8 bioconda::iqtree=2.1.4_beta
  status : 1
  message:
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    
    # >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
    
        Traceback (most recent call last):
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/conda/exceptions.py", line 1080, in __call__
            return func(*args, **kwargs)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 951, in exception_converter
            raise e
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 945, in exception_converter
            exit_code = _wrapped_main(*args, **kwargs)
executor >  local (1)
[ad/44ede2] process > BACTOPIATOOLS:PANGENOME:PG_TOOL:PIRATE_MODULE (pirate) [100%] 1 of 1 ✔
[-        ] process > BACTOPIATOOLS:PANGENOME:SNPDISTS                       -
[-        ] process > BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:IQTREE           -
[-        ] process > BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:CLONALFRAME      -
[-        ] process > BACTOPIATOOLS:PANGENOME:FINAL_TREE:IQTREE_MODULE       -
[-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS              -
Creating env using mamba: bioconda::iqtree=2.1.4_beta [cache /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8]
Creating env using mamba: bioconda::snp-dists=0.8.2 [cache /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-e13ce3aa35d34f1f090a292512579027]
Error executing process > 'BACTOPIATOOLS:PANGENOME:CLONALFRAMEML:IQTREE (start-tree)'

Caused by:
  Failed to create Conda environment
  command: mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8 bioconda::iqtree=2.1.4_beta
  status : 1
  message:
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    error    libmamba Could not set lock (Resource temporarily unavailable)
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    warning  libmamba Cannot lock '/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache'
        Waiting for other mamba process to finish
    
    # >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
    
        Traceback (most recent call last):
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/conda/exceptions.py", line 1080, in __call__
            return func(*args, **kwargs)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 951, in exception_converter
            raise e
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 945, in exception_converter
            exit_code = _wrapped_main(*args, **kwargs)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 901, in _wrapped_main
            result = do_call(args, p)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 767, in do_call
            exit_code = create(args, parser)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 643, in create
            return install(args, parser, "create")
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/mamba.py", line 507, in install
            index = load_channels(pool, channels, repos)
          File "/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/python3.7/site-packages/mamba/utils.py", line 171, in load_channels
            repo = subdir.create_repo(pool)
        RuntimeError: Could not read JSON repodata file (/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs/cache/31ce02e0.json) parse error line 163727
    
    `$ /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/bin/mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/share/bactopia-2.0.x/conda/envs/env-79584ec670f934938876952992662fb8 bioconda::iqtree=2.1.4_beta`
    
      environment variables:
                     CIO_TEST=<not set>
       CONDA_BACKUP_JAVA_HOME=:-:-
    CONDA_BACKUP_JAVA_LD_LIBRARY_PATH=:-:-
            CONDA_DEFAULT_ENV=bactopia-dev
                    CONDA_EXE=/Users/jvanderwoude3/miniconda3/bin/conda
                 CONDA_PREFIX=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev
               CONDA_PREFIX_1=/Users/jvanderwoude3/miniconda3
        CONDA_PROMPT_MODIFIER=(bactopia-dev)
             CONDA_PYTHON_EXE=/Users/jvanderwoude3/miniconda3/bin/python
                   CONDA_ROOT=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev
                  CONDA_SHLVL=2
               CURL_CA_BUNDLE=<not set>
         JAVA_LD_LIBRARY_PATH=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/lib/server
                         PATH=/Users/jvanderwoude3/miniconda3/envs/bactopia-dev/bin:/Users/jvanderwo
                              ude3/miniconda3/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:
                              /usr/local/ncbi/blast/bin:/opt/salt/bin:/usr/local/sbin:/opt/X11/bin:/
                              Users/jvanderwoude3/edirect
           REQUESTS_CA_BUNDLE=<not set>
                SSL_CERT_FILE=<not set>
    
         active environment : base
        active env location : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev
                shell level : 2
           user config file : /Users/jvanderwoude3/.condarc
     populated config files : /Users/jvanderwoude3/.condarc
              conda version : 4.11.0
        conda-build version : not installed
             python version : 3.7.12.final.0
           virtual packages : __osx=10.16=0
                              __unix=0=0
                              __archspec=1=x86_64
           base environment : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev  (writable)
          conda av data dir : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/etc/conda
      conda av metadata url : None
               channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                              https://conda.anaconda.org/conda-forge/noarch
                              https://conda.anaconda.org/bioconda/osx-64
                              https://conda.anaconda.org/bioconda/noarch
                              https://repo.anaconda.com/pkgs/main/osx-64
                              https://repo.anaconda.com/pkgs/main/noarch
                              https://repo.anaconda.com/pkgs/r/osx-64
                              https://repo.anaconda.com/pkgs/r/noarch
                              https://conda.anaconda.org/r/osx-64
                              https://conda.anaconda.org/r/noarch
              package cache : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/pkgs
                              /Users/jvanderwoude3/.conda/pkgs
           envs directories : /Users/jvanderwoude3/miniconda3/envs/bactopia-dev/envs
                              /Users/jvanderwoude3/.conda/envs
                   platform : osx-64
                 user-agent : conda/4.11.0 requests/2.27.1 CPython/3.7.12 Darwin/20.6.0 OSX/10.16
                    UID:GID : 1041747584:536040563
                 netrc file : None
               offline mode : False
    
    
    An unexpected error has occurred. Conda has prepared the above report.




Waiting files transfer to complete (15 files)
WARN: To render the execution DAG in the required format it is required to install Graphviz -- See http://www.graphviz.org for more info.

**Comment by rpetit3** [comment]
Hmmm, for some reason it put you back to v2.0.0 of Bactopia. The issue is conda envs being built in parallel, which was fixed in v2.0.2.

Let's try this, but being explicit this time can force v2.0.2
```
conda clean -ay
conda create -n bactopia-dev -c rpetit3 -c conda-forge -c bioconda 'rpetit3::bactopia=2.0.2'
```

It will ask if you want to overwrite the existing environment, you can say yes.

**Comment by jellyvanderwoude** [comment]
Trying this code returns an error, hmm. Thanks for helping me chase this down!

> Remove existing environment (y/[n])? y
> 
> Collecting package metadata (current_repodata.json): done
> Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
> Collecting package metadata (repodata.json): done
> Solving environment: failed
> 
> PackagesNotFoundError: The following packages are not available from current channels:
> 
>   - rpetit3::bactopia=2.0.2
> 
> Current channels:
> 
>   - https://conda.anaconda.org/rpetit3/osx-64
>   - https://conda.anaconda.org/rpetit3/noarch
>   - https://conda.anaconda.org/conda-forge/osx-64
>   - https://conda.anaconda.org/conda-forge/noarch
>   - https://conda.anaconda.org/bioconda/osx-64
>   - https://conda.anaconda.org/bioconda/noarch
>   - https://repo.anaconda.com/pkgs/main/osx-64
>   - https://repo.anaconda.com/pkgs/main/noarch
>   - https://repo.anaconda.com/pkgs/r/osx-64
>   - https://repo.anaconda.com/pkgs/r/noarch
>   - https://conda.anaconda.org/r/osx-64
>   - https://conda.anaconda.org/r/noarch
> 
> To search for alternate channels that may provide the conda package you're
> looking for, navigate to
> 
>     https://anaconda.org
> 
> and use the search bar at the top of the page.

**Comment by rpetit3** [comment]
Oh! The issue here is I don't have a dev build for Mac, just Linux.

We can manually fix this if you like

**Comment by jellyvanderwoude** [comment]
If it's not too much trouble, sure. Otherwise, I can wait for the update. Let me know what works for you.

**Comment by rpetit3** [comment]
You will want to replace this file

```
/Users/jvanderwoude3/miniconda3/envs/bactopia/share/bactopia-2.0.x/lib/WorkflowBactopiaTools.groovy
```

With this file https://github.com/bactopia/bactopia/raw/master/lib/WorkflowBactopiaTools.groovy (right click save as)

Then you can use your original `bactopia` environment

**Comment by jellyvanderwoude** [comment]
Worked like a charm!

**Comment by marcelladane** [comment]
> 

Hey, when is the new version that has this bug fixed going to be up? =)

**Comment by rpetit3** [comment]
@marcelladane I was thinking it was about time to push a version update. I'll go ahead an get that process started, and get an update out today

**Comment by marcelladane** [comment]
Thank you very much.  :)

**Comment by rpetit3** [comment]
Hi @jellyvanderwoude  and @marcelladane 

I've released v2.0.3 (https://github.com/bactopia/bactopia/releases/tag/v2.0.3) of Bactopia, which includes this fix. It should be available from Bioconda today or tomorrow.  I'm going to go ahead and close this, but please feel free to reopen.

Thank you so much for helping to find this bug! Your feedback has been super helpful!
Robert
